app/uk/gov/hmrc/helptosave/services/UserInfoService.scala

```diff
@@ -22,40 +22,40 @@ import cats.syntax.cartesian._
 import cats.syntax.option._
 import com.google.inject.Inject
 import uk.gov.hmrc.helptosave.connectors.CitizenDetailsConnector.CitizenDetailsResponse
-import uk.gov.hmrc.helptosave.connectors.{CitizenDetailsConnector, UserDetailsConnector}
 import uk.gov.hmrc.helptosave.connectors.UserDetailsConnector.UserDetailsResponse
-import uk.gov.hmrc.helptosave.models.{Address, UserInfo}
-import uk.gov.hmrc.helptosave.util.{NINO, Result}
+import uk.gov.hmrc.helptosave.connectors.{CitizenDetailsConnector, UserDetailsConnector}
+import uk.gov.hmrc.helptosave.models._
+import uk.gov.hmrc.helptosave.util.{EligibilityResultType, NINO, Result}
 import uk.gov.hmrc.play.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
 
-
 class UserInfoService @Inject()(userDetailsConnector: UserDetailsConnector,
                                 citizenDetailsConnector: CitizenDetailsConnector) {
 
-  def getUserInfo(userDetailsUri: String, nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext): Result[UserInfo] =
+  def getUserInfo(userDetailsUri: String, nino: NINO)(implicit hc: HeaderCarrier, ec: ExecutionContext):
+  EligibilityResultType[UserInfo] =
     for {
-      userDetails ← userDetailsConnector.getUserDetails(userDetailsUri)
-      citizenDetails ← citizenDetailsConnector.getDetails(nino)
-      userInfo ← EitherT.fromEither[Future](toUserInfo(userDetails, citizenDetails, nino).toEither).leftMap(
-        errors ⇒ s"Could not create user info: ${errors.toList.mkString(",")}")
+      userDetails ← userDetailsConnector.getUserDetails(userDetailsUri).leftMap(e ⇒ HtsError(UnKnown, e))
+      citizenDetails ← citizenDetailsConnector.getDetails(nino).leftMap(e ⇒ HtsError(UnKnown, e))
+      userInfo ← EitherT.fromEither[Future](toUserInfo(userDetails, citizenDetails, nino).toEither)
+        .leftMap(errors ⇒ HtsError(UserInfoType, errors.toList.mkString(",")))
     } yield userInfo
 
   private def toUserInfo(u: UserDetailsResponse,
                          c: CitizenDetailsResponse,
                          nino: NINO): ValidatedNel[String, UserInfo] = {
     val surnameValidation =
       u.lastName.orElse(c.person.flatMap(_.lastName))
-        .toValidNel("Could not find last name")
+        .toValidNel("Surname")
```

Comment:
Returning a `String` is probably not best here if those strings have specific meanings which influence behaviour later on - you have to look through the whole function to find out what the different strings can be. You could probably have the strings be part of some enum, but it is easier to see with the types if you create a sealed trait such as:
```scala
sealed trait UserInfoError
case object SurnameMissing extends UserInfoError
case object DateOfBirthMissing extends UserInfoError
...
```
It is easier to see what the errors can be in this case. Previously we were just logging the strings so we didn't care